vendor/sigs.k8s.io/controller-runtime/pkg/client/fake/client.go

```diff
@@ -110,7 +111,48 @@ func (c *fakeClient) List(ctx context.Context, opts *client.ListOptions, list ru
 	}
 	decoder := scheme.Codecs.UniversalDecoder()
 	_, _, err = decoder.Decode(j, nil, list)
-	return err
+	if err != nil {
+		return err
+	}
+
+	// TODO: can be removed after vendoring the controller-runtime with the PR
+	// https://github.com/kubernetes-sigs/controller-runtime/pull/311
```

Comment:
Backporting #311 into release-0.1 branch: https://github.com/kubernetes-sigs/controller-runtime/pull/410.

Reply:
Good job, thanks! Do I need to wait until this one will be merged?

Reply:
We might temporarily fork `controller-runtime` repo and point Godep.toml to it. Otherwise, there's a high chance your patch will get forgotten during the next deps sync of the project and your changes will get undone.

Reply:
Please, use https://github.com/ingvagabund/controller-runtime/tree/release-0.1-backport-311 instead until https://github.com/kubernetes-sigs/controller-runtime/pull/410 gets merged. Plus, commenting why we have it instead of the upstream one.

Reply:
done